recipe/meta.yaml

```diff
@@ -15,18 +16,40 @@ source:
     - 0202-Fix-for-cuda-compute-capabilities-setting.patch  #[build_type == 'cuda']
 
 build:
-  number: 4
+  number: 1
+  run_exports:
+    - {{ pin_subpackage("liblightgbm", max_pin="x.x.x") }}
 
 outputs:
-  - name: liblightgbm-base
+  - name: lightgbm-proc
+    version: {{ build_ext_version }}
+    build:
+      number: {{ proc_build_number }}
+      string: {{ build_type }}
+    test:
+      commands:
+         - exit 0
+    about:
+      home: https://github.com/microsoft/LightGBM
+      license: MIT License
+      license_family: MIT
+      license_file: LICENSE
+      summary: Light Gradient Boosting Machine that uses tree based learning algorithms
+
+  - name: liblightgbm
     script: install-liblightgbm.sh
+    version: {{ version }}
     build:
-      string: {{ build_type }}_{{ mpi_type }}_{{ PKG_BUILDNUM }} #[build_type == 'cpu']
-      string: {{ build_type }}{{ cudatoolkit | replace(".*", "") }}_{{ mpi_type }}_{{ PKG_BUILDNUM }} #[build_type == 'cuda']
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ build_type }}_{{ mpi_type }} #[build_type == 'cpu']
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ build_type }}{{ cudatoolkit | replace(".*", "") }}_{{ mpi_type }} #[build_type == 'cuda']
 {% if build_type == 'cuda' %}
       script_env:
         - CUDA_HOME
 {% endif %}
+      run_exports:
+        - {{ pin_subpackage("liblightgbm", max_pin="x.x.x") }}
```

Comment:
Why is this being added?

Reply:
Yeah, that seems like it pins to itself.
We have a similar line in the arrow recipe which is mostly what this update is based on.
https://github.com/open-ce/pyarrow-feedstock/blob/main/recipe/meta.yaml#L46-L47

I'll try to remove that and see what happens.

Reply:
It seems to work just fine without this explicit run_export.
I'll remove it.